app/Http/Controllers/Assets/BulkAssetsController.php

```diff
@@ -30,9 +31,17 @@ public function edit(Request $request)
         $this->authorize('update', Asset::class);
 
         if (! $request->filled('ids')) {
-            return redirect()->back()->with('error', 'No assets selected');
+            return redirect()->back()->with('error', trans('admin/hardware/message.update.no_assets_selected'));
+
         }
 
+        // Figure out where we need to send the user after the update is complete, and store that in the session
+        $bulk_back_url = request()->headers->get('referer');
+        session(['bulk_back_url' => $bulk_back_url]);
```

Comment:
I wonder if this might actually work better as a URL parameter - I'm worried about what might happen if someone opens a few bulk-edit windows at the same time, then submits them out-of-order - the last 'bulk_back_url' would 'win' I believe?

But I don't think that's fatal, and I junky URL's with query params in them can be gross, so maybe not...

Reply:
How do you mean as a URL parameter?

Reply:
e.g. `hardware/bulk_edit?from=hardware.index` - or maybe there's an even nicer way to do it like `hardware/bulk_edit/hardware.index` or, since `hardware.index` ought to be the default, just `hardware/bulk_edit` but others being `hardware/bulk_edit/audit.index` or something like that?

Reply:
Those URLs look awful tho, and we'd need to make sure we allow-listed the parameters so people can't fuzz them.

Reply:
Yup, they do. And, I mean, you can go ahead and fuzz them but you could also just browse to them; they'd never let you go anywhere you couldn't go already. But they do look gross, and that was one of my concerns.